libbpf-rs/src/program.rs

```diff
@@ -327,4 +327,15 @@ impl Program {
             Ok(Link::new(ptr))
         }
     }
+
+    /// Attach a verdict/parser to a [sockmap/sockhash](https://lwn.net/Articles/731133/)
+    pub fn attach_sockmap(&self, map_fd: i32) -> Result<()> {
```

Comment:
Does it work to pass a map? Would be nicer to do a Map object if it doesn't cause borrowing issues (need 2 mutable borrows from Object)
```suggestion
    pub fn attach_sockmap(&self, map: &Map) -> Result<()> {
```

Reply:
It would only work if `Map` implements `Clone` which is not the case now. I don't know enough about the project to see whether this is OK.

Here is a potential code snippet with `clone()`:

```rust
        let sockmap = obj.map("sock_map")?.ok_or("unknown map: sock_map")?.clone();
        obj.prog("handle__verdict")?
            .ok_or("unknown program: handle__verdict")?
            .attach_sockmap(&sockmap)?;
        obj.prog("handle__parser")?
            .ok_or("unknown program: handle__parser")?
            .attach_sockmap(&sockmap)?;
```

The previous one I used with the file descriptor:

```rust
        let sockmap = obj.map("sock_map")?.ok_or("unknown map: sock_map")?.fd();
        obj.prog("handle__verdict")?
            .ok_or("unknown program: handle__verdict")?
            .attach_sockmap(sockmap)?;
        obj.prog("handle__parser")?
            .ok_or("unknown program: handle__parser")?
            .attach_sockmap(sockmap)?;
```

Let me know which one is best and I'll adapt my PR.

Reply:
`Clone` for `Map` seems dangerous b/c a cloned Map would alias the underlying pointer and potentially cause safety issues. Let's just go with the fd-based approach b/c I can't think of anything better right now